src/main/scala/sigmastate/eval/RuntimeCosting.scala

```diff
@@ -1155,6 +1155,9 @@ trait RuntimeCosting extends SigmaLibrary with DataCosting with Slicing { IR: Ev
 
       case AtLeast(bound, input) =>
         val inputC = asRep[CostedCol[Any]](evalNode(ctx, env, input))
+        val inputCount = inputC.values.length.asValue
```

Comment:
@greenhat This usage of `asValue` is not always safe. It will only work if collection is constructed by using `fromItems` method. However this will not always be the case. Check out how this function is implemented and extract constant only when there is a constant. Otherwise do nothing.